include/boost/url/impl/const_string.hpp

```diff
@@ -172,9 +172,10 @@ const_string(
     string_view s,
     Allocator const& a)
 {
-    // VFALCO Should we do something special when n==0?
     if (is_small(s.size()))
     {
+        if (s.empty())
+            s = "";
```

Comment:
why do this?

why not `s = {};`?

Reply:
`s = {}` was what caused the problem in the first place. `s.p_` becomes `nullptr`. Anyway, we don't need this line anymore.

```cpp
BOOST_CONSTEXPR basic_string_view() BOOST_NOEXCEPT: p_(), n_()
    {
    }
```